### PR TITLE
Whats new in 1.0

### PR DESCRIPTION
**Pull request by wafels**
### Description

A file describing what's new in 1.0

Fixes #<Issue Number>

### CODE REVIEW

File: docs/whatsnew/1.0.rst [hunk 1]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
...
 
-
 SunPy is tested against Python 3.6, and 3.7.
 
+.. _whatsnew-1.0-time:
+
+Astropy Time is used everywhere
+===============================
+
+SunPy now uses Astropy's Time object everywhere to represent time.  For a full
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion

SunPy now uses Astropy's `~astropy.time.Time` object everywhere to represent time.

This comes with numerous benefits:

- **Support for non-UTC time scales.**

  UTC as well as non-UTC time scales like TAI, TT, UT1 etc. can be used with `astropy.time.Time`. 

  .. code:: python

      >>> t = Time('2012-06-18T02:00:05.453', scale='tai')
      >>> t
      <Time object: scale='tai' format='isot' value=2012-06-18T02:00:05.453>

  `~astropy.time.Time` also provides easy conversion between different scales.

  .. code:: python

      >>> t.utc
      <Time object: scale='utc' format='isot' value=2012-06-18T01:59:31.453>

- **Support for high precision times.**

  `~astropy.time.Time` can provide sub-nanosecond precision for time objects while python
  `datetime` was restricted to microseconds.

  .. code:: python

    >>> t = Time('2012-06-18T02:00:05.453123123')
    >>> t
    <Time object: scale='utc' format='isot' value=2012-06-18T02:00:05.453>
    >>> t.precision = 9
    >>> t
    <Time object: scale='utc' format='isot' value=2012-06-18T02:00:05.453123123>

- **Support for leap seconds**

  This was one of the bigger motivation for the transition to `astropy.time.Time`.
  `datetime` has no support for leap second while `~astropy.time.Time` supports leap seconds.
  A leap second is a one-second adjustment applied to UTC to keep it close to the mean solar time.

  .. code:: python

    >>> Time('2016-12-31T23:59:60')
    <Time object: scale='utc' format='isot' value=2016-12-31T23:59:60.000>

    >>> Time('2016-12-31T23:59:59') + 1 * u.s
    <Time object: scale='utc' format='isot' value=2016-12-31T23:59:60.000>

- **Support for numerous formats**

  `~astropy.time.Time` can parse numerous formats including python `datetime`.

  .. code:: python

    >>> list(Time.FORMATS)
    ['jd', 'mjd', 'decimalyear', 'unix', 'cxcsec', 'gps', 'plot_date', 'datetime', 'iso', 'isot', 'yday', 'fits', 'byear', 'jyear', 'byear_str', 'jyear_str'] 

  .. code:: python
    
    >>> import datetime
    >>> Time(datetime.datetime.now())
    <Time object: scale='utc' format='datetime' value=2018-10-20 15:36:16.364089>

Changes to SunPy
++++++++++++++++

- **All functions which used to return** `datetime` **now return** `~astropy.time.Time`

  And all functions which return `datetime.timedelta` now return `astropy.time.TimeDelta`.
  For example, the properties of `sunpy.time.TimeRange` which  used to return `datetime.datetime` and `datetime.timedelta` now return `astropy.time.Time` and `astropy.time.TimeDelta`

- **Changes to** `~sunpy.time.parse_time`

  `~sunpy.time.parse_time` has been reduced to a tiny wrapper over `~astropy.time.Time`. 
  The API of `~sunpy.time.parse_time` is mostly similar to that of `~astropy.time.Time`. 
  `~sunpy.time.parse_time` supports parsing a few more formats than `~astropy.time.Time`, which
  are `numpy.datetime64`, `pandas.Series`, `pandas.DatetimeIndex`, utime and a few other time formats.

The work on the transition from `datetime` to `astropy.time.Time` was done as a part of  Vishnunarayan's Google Summer of Code 2018 project.
```

File: docs/whatsnew/1.0.rst [hunk 2]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
...
+There have been numerous improvements to large parts of SunPy, notably in the content of SunPy's documentation, and the continuous integration testing of SunPy.
+In addition, SunPy co-ordinate system functionality has been improved, and the transformation between SunPy specific co-ordinate systems and those implemented by Astropy is now tested.
+
+.. _whatsnew-1.0-python:
 
 Supported versions of Python
 ============================
 
-
 SunPy is tested against Python 3.6, and 3.7.
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion
SunPy 1.0 has dropped Python 2 support as Python 2 is nearing the end of its support cycle.
To port your code to Python 3, there is some good advice `here <https://docs.python.org/3/howto/pyporting.html>`__.
If you need Python 2 support, you use SunPy 0.9.X which will see bug fixes until 2020. 

As a result, SunPy's minimum Python version is Python 3.6 and is routinely against Python 3.6, and 3.7.
```

File: docs/whatsnew/1.0.rst [hunk 3]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
 Overview
 ========
 
-SunPy 1.0 brings <Fill in>
+SunPy 1.0 now uses Astropy Time everywhere
+This and all future versions of SunPy will support only Python 3.6 and higher.
+This is because Python 2 `will not be maintained after 2019 <https://python3statement.org/>`_.
+The 0.9.x series will receive bugfixes only up until the end of life of Python 2 (around 6 months).
+No new functionality will be added to the 0.9.x series, which will also be the last version to include `sunpy.spectra`, `sunpy.lightcurve` and `sunpy.wcs`, all of which were deprecated in 0.8.
+All new functionality will be available only in SunPy 1.0 and higher.
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion
```

File: docs/whatsnew/1.0.rst [hunk 4]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
 Overview
 ========
 
-SunPy 1.0 brings <Fill in>
+SunPy 1.0 now uses Astropy Time everywhere
+This and all future versions of SunPy will support only Python 3.6 and higher.
+This is because Python 2 `will not be maintained after 2019 <https://python3statement.org/>`_.
+The 0.9.x series will receive bugfixes only up until the end of life of Python 2 (around 6 months).
+No new functionality will be added to the 0.9.x series, which will also be the last version to include `sunpy.spectra`, `sunpy.lightcurve` and `sunpy.wcs`, all of which were deprecated in 0.8.
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion
```

File: docs/whatsnew/1.0.rst [hunk 5]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
 Overview
 ========
 
-SunPy 1.0 brings <Fill in>
+SunPy 1.0 now uses Astropy Time everywhere
+This and all future versions of SunPy will support only Python 3.6 and higher.
+This is because Python 2 `will not be maintained after 2019 <https://python3statement.org/>`_.
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion
```

File: docs/whatsnew/1.0.rst [hunk 6]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
 Overview
 ========
 
-SunPy 1.0 brings <Fill in>
+SunPy 1.0 now uses Astropy Time everywhere
+This and all future versions of SunPy will support only Python 3.6 and higher.
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion
```

File: docs/whatsnew/1.0.rst [hunk 7]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
 Overview
 ========
 
-SunPy 1.0 brings <Fill in>
+SunPy 1.0 now uses Astropy Time everywhere
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion
```

File: docs/whatsnew/1.0.rst (hunk 1 above)
Comment (Cadair) [reply docs/whatsnew/1.0.rst]: my goodness that's a long suggestion.

File: docs/whatsnew/1.0.rst (hunk 7 above)
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion

SunPy 1.0 brings these headline changes:

* A complete transition of the whole code base to use ~astropy.time.Time, which was implemented by Vishnunarayan K I as part of Google Summer of Code 2018.
* A rewrite of how all the clients in sunpy.net download files from the internet. This means vastly improved progress bars, skipping downloads if files are present, and better visibility and retrying of download errors.
* A rewrite of the differential rotation and image warping code to correctly account for observer location using the Astropy coordinate functionality.
* Removal of many deprecated functions and submodules, we have used the 1.0 release as a chance to clean out SunPy reducing the number of lines of Python code in the project by almost 3,000!
* The first release of SunPy to be Python 3 only, requiring Python 3.6+.
```

File: docs/whatsnew/1.0.rst [hunk 8]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
...
+
+* 1691 commits have been added since 0.9
+* 546 issues have been closed since 0.9
+* ??? pull requests have been merged since 0.9
+* 48 people have contributed since 0.9
+* 21 new contributors
+
+There are numerous breaking changes to SunPy, notably in the implementation of time, which has been switched to using Astropy yimr
+There have been numerous improvements to large parts of SunPy, notably in the content of SunPy's documentation, and the continuous integration testing of SunPy.
+In addition, SunPy co-ordinate system functionality has been improved, and the transformation between SunPy specific co-ordinate systems and those implemented by Astropy is now tested.
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion
```

File: docs/whatsnew/1.0.rst [hunk 9]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
...
+By the numbers:
+
+* 1691 commits have been added since 0.9
+* 546 issues have been closed since 0.9
+* ??? pull requests have been merged since 0.9
+* 48 people have contributed since 0.9
+* 21 new contributors
+
+There are numerous breaking changes to SunPy, notably in the implementation of time, which has been switched to using Astropy yimr
+There have been numerous improvements to large parts of SunPy, notably in the content of SunPy's documentation, and the continuous integration testing of SunPy.
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion
```

File: docs/whatsnew/1.0.rst [hunk 10]
```diff
@@ -9,14 +9,66 @@ What's New in SunPy 1.0?
...
+
+By the numbers:
+
+* 1691 commits have been added since 0.9
+* 546 issues have been closed since 0.9
+* ??? pull requests have been merged since 0.9
+* 48 people have contributed since 0.9
+* 21 new contributors
+
+There are numerous breaking changes to SunPy, notably in the implementation of time, which has been switched to using Astropy yimr
```
Comment (nabobalis) [inline docs/whatsnew/1.0.rst]:
```suggestion
```

File: docs/whatsnew/1.0.rst (hunk 1 above)
Comment (wafels) [reply docs/whatsnew/1.0.rst]: ok

**Comment by Cadair** [comment]
> /root/project/docs/whatsnew/1.0.rst:68:Unexpected indentation.

Would be good to get the build passing on this so we can see the preview.

**Comment by nabobalis** [comment]
i'm pretty sure I never put a config file in my Library path because I never even knew where that directory was.
in the swtich was there a move of the old config file to the new path?
Cadair
I am almost 100% sure we never write to the users config files
ehsteve
in the swtich was there a move of the old config file to the new path?
I don't think we account for this at all

File: docs/whatsnew/1.0.rst [hunk 11]
```diff
@@ -9,18 +9,275 @@ What's New in SunPy 1.0?
...
+.. _whatsnew-1.0-renamed-removed:
 
 Renamed/removed functionality
 =============================
 
-<Fill In>
+This is just some of the renamed or removed functionality.
+
+* ``sunpy.sun.sun`` functions have been re-implemented using Astropy for significantly improved accuracy. Some functions have been removed.
+* Renamed ``eccentricity_sunearth_orbit`` to ``eccentricity_sun_earth_orbit``.
```
Comment (ayshih) [inline docs/whatsnew/1.0.rst]:
```suggestion
```
I subsequently deleted this function, so it's probably not worth mentioning here.

**Comment by Cadair** [comment]
The doc builds are passing again (HEK is back) so these fails are real.

**Comment by nabobalis** [comment]
Who broke the file?

**Pull request merged**